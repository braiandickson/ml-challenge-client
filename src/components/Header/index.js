import React from 'react';
import { useHistory } from 'react-router-dom';

import { useDispatch } from '../../contexts/ProductsContext';
import { actionCreators } from '../../contexts/ProductsContext/actions';
import logo from '../../assets/Header/logo.png';
import PATHS from '../Routes/paths';

import SearchBar from '../SearchBar';

const Header = () => {
  const dispatch = useDispatch();
  const history = useHistory();
  const handleSearch = (query) => {
    dispatch(actionCreators.getProducts(query, dispatch));
    history.push(`${PATHS.results}?search=${query}`);
  };
  return (
    <div className="row center full-width header">
      <div className="row middle space-between header__container">
        <img src={logo} alt="logo" className="header__logo" />
        <SearchBar handleSearch={handleSearch} />
      </div>
    </div>
  );
};

export default Header;
