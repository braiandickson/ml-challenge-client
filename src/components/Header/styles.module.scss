@import '../../styles/variables/colors';

.header {
  background-color: $yellow;
  height: 60px;

  &__container {
    max-width: 1200px;
  }

  &__button {
    background-color: $yellow;
    cursor: pointer;
    margin-right: 60px;

    &:focus {
      outline: none;
    }
  }

  &__logo {
    height: 34px;
    width: 134px;
  }
}
