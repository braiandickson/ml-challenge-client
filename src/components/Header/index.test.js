import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Header from './index';

describe('#Header', () => {
  it('Mounts the Header component correctly with data', () => {
    const component = mount(<Header />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
