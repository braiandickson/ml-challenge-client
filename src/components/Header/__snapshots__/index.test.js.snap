// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Header Mounts the Header component correctly with data 1`] = `
<Header>
  <div
    className="row center full-width header"
  >
    <div
      className="row middle space-between header__container"
    >
      <button
        className="header__button"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="logo"
          className="header__logo"
          src="logo.png"
        />
      </button>
      <SearchBar
        handleSearch={[Function]}
      >
        <form
          className="row middle full-width searchbar"
          onSubmit={[Function]}
        >
          <input
            className="searchbar__input"
            name="query"
            onChange={[Function]}
            type="text"
          />
          <button
            className="searchbar__button"
            type="submit"
          >
            <img
              alt="search-icon"
              className="searchbar__icon"
              src="searchIcon.svg"
            />
          </button>
        </form>
      </SearchBar>
    </div>
  </div>
</Header>
`;
